### DISCUSSION TITLE
disable OneAgent via Monaco

### DISCUSSION THREAD

**Discussion by jodycroesrd** [Dynatrace/dynatrace-configuration-as-code#270] [Archive] [answered]
Hello,

is it possible to disable OneAgents on specific hosts via Monaco ("monitoringEnabled": false,)?
It's possible via Rest API (configuration), however at this moment i don't see it in list of supported configurations ...

Thanks in advance,
Jody

**Comment by tatomir146** [comment, marked as answer]
Hi @jodycroesrd, this is still not supported in monaco. We would be happy about any contribution to this project. [Here](https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/blob/main/docs/how-to-add-a-new-api.md) you can find more information about adding new APIs.

**Reply by jodycroesrd** [reply]
after some more investigation it seems that the API does not meet the requirements in order to be added to Monaco ...

- It implements the following HTTP methods. E.g for configuration APIs that is:
GET <my-environment>/api/config/v1/<my-config> (get all configs)
GET <my-environment>/api/config/v1/<my-config>/<id> (get a single config)
POST <my-environment>/api/config/v1/<my-config> (create a new config)
PUT <my-environment>/api/config/v1/<my-config>/<id> (change an existing config)
DELETE <my-environment>/api/config/v1/<my-config>/<id> (delete a config)

- The model of the configuration has a name property:
{
    "id": "acbed0c4-4ef1-4303-991f-102510a69322",
    "name: "my-name"
    ...
}

When using / checking the Condig API:
/hosts/{id}/monitoring Gets the monitoring configuration of OneAgent on the specified host
>> we are lacking name property

{
  "metadata": {
    "configurationVersions": [
      9
    ],
    "clusterVersion": "1.212.111.20210322-210138"
  },
  "id": "HOST-858B2B7273B020B5",
  "monitoringEnabled": false,
  "monitoringMode": "FULL_STACK"
}

Any chance to get this supported?

**Reply by tatomir146** [reply]
After releasing monaco 2.0 we will work on supporting these kind of API endpoints.

**Reply by jodycroes** [reply]
Hello Sinisa,

since we are still at version 1.6, can you let me know when v2.0 will be released and as a result disabling monitoring will be supported?

Regards,
Jody

**Reply by jodycroes** [reply]
Hello Sinisa,

any update on this as we still see at v 1.7

Thanks,
Jody
